source/full_shutdown.rst

```diff
@@ -0,0 +1,205 @@
+.. include:: vars.rst
+
+=======================
+Full Shutdown Procedure
+=======================
+
+In case a full shutdown of the system is required, we advise to use the
+following order:
+
+* Perform a graceful shutdown of all virtual machine instances
+* Stop Ceph (if applicable)
+* Put all nodes into maintenance mode in Bifrost
+* Shut down compute nodes
+* Shut down monitoring node
+* Shut down network nodes (if separate from controllers)
+* Shut down controllers
+* Shut down Ceph nodes (if applicable)
+* Shut down seed VM
+* Shut down Ansible control host
+
+Virtual Machines shutdown
+-------------------------
+
+Contact Openstack users to stop their virtual machines gracefuly, 
+If that is not possible shut down VMs using openstack CLI as admin user:
+
+.. code-block:: bash
+
+   for i in `openstack server list --all-projects -c ID -f value` ; \
+   do openstack server stop $i ; done
+
+
+.. ifconfig:: deployment['ceph_managed']
+
+   Stop Ceph
+   ---------
+   Procedure based on `Red Hat documentation <https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/4/html/administration_guide/understanding-process-management-for-ceph#powering-down-and-rebooting-a-red-hat-ceph-storage-cluster_admin>`__ 
+
+   - Stop the Ceph clients from using any Ceph resources (RBD, RADOS Gateway, CephFS)
+   - Check if cluster is in healthy state
+
+   .. code-block:: bash
+
+      ceph status 
+
+   - Stop CephFS (if applicable)
+
+   Stop CephFS cluster by reducing the number of ranks to 1, setting the cluster_down flag, and then failing the last rank.
+
+   .. code-block:: bash
+
+      ceph fs set FS_NAME max_mds 1
+      ceph mds deactivate FS_NAME:1 # rank 2 of 2
+      ceph status # wait for rank 1 to finish stopping
+      ceph fs set FS_NAME cluster_down true
+      ceph mds fail FS_NAME:0
+
+   Setting the cluster_down flag prevents standbys from taking over the failed rank.
+
+   - Set the noout, norecover, norebalance, nobackfill, nodown and pause flags.
+
+   .. code-block:: bash
+
+      ceph osd set noout
+      ceph osd set norecover
+      ceph osd set norebalance
+      ceph osd set nobackfill
+      ceph osd set nodown
+      ceph osd set pause
+
+   - Shut down the OSD nodes one by one:
+
+   .. code-block:: bash
+
+      systemctl stop ceph-osd.target
+
+   - Shut down the monitor/manager nodes one by one:
+
+   .. code-block:: bash
+
+      systemctl stop ceph.target
+
+Set Bifrost maintenance mode
+----------------------------
+
+Set maintenance mode in bifrost to prevent nodes from automatically
+powering back on
+
+.. code-block:: bash
+
+   for i in `openstack --os-cloud bifrost baremetal node list -c UUID -f value` ; \ 
+   do openstack --os-cloud bifrost baremetal node maintenance set $i ; done
+
+Shut down nodes
+---------------
+
+Shut down nodes one at a time gracefully using:
+
+.. code-block:: bash
+
+   systemctl poweroff
```

Comment:
There might be serialised form of shutdown invocation using Kayobe's tools https://docs.openstack.org/kayobe/latest/administration/overcloud.html#running-commands - perhaps also with a small delay to the shutdown command so that it doesn't immediately chop off the ansible connection.